### DISCUSSION TITLE
URC4450BC0-X-R, XHK1-EU Zigbee Keypad support?

### DISCUSSION THREAD

**Discussion by Multipage** [Koenkk/zigbee2mqtt#5822] [Devices]
When will this device will be supported?
This topic on reddit has all the info:
https://www.reddit.com/r/homeassistant/comments/k1hg4w/physical_zigbee_alarm_keypad_integrated_into_home/

I've allready tried using
`zigbee_shepherd_devices: true`
And used the devices.js in my : /usr/share/hassio/share/zigbee2mqtt

But then it won't start:
`2021-01-19T14:04:19: PM2 log: App [npm:0] exited with code [1] via signal [SIGINT]
2021-01-19T14:04:19: PM2 log: App [npm:0] starting in -fork mode-
2021-01-19T14:04:19: PM2 log: App [npm:0] online
> zigbee2mqtt@1.17.0 start /zigbee2mqtt-1.17.0
> node index.js
internal/modules/cjs/loader.js:968
  throw err;
  ^
Error: Cannot find module './converters/common'
Require stack:
- /zigbee2mqtt-1.17.0/node_modules/zigbee-herdsman-converters/devices.js
- /zigbee2mqtt-1.17.0/node_modules/zigbee-herdsman-converters/index.js
- /zigbee2mqtt-1.17.0/lib/zigbee.js
- /zigbee2mqtt-1.17.0/lib/controller.js
- /zigbee2mqtt-1.17.0/index.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:965:15)
    at Function.Module._load (internal/modules/cjs/loader.js:841:27)
    at Module.require (internal/modules/cjs/loader.js:1025:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/zigbee2mqtt-1.17.0/node_modules/zigbee-herdsman-converters/devices.js:23:16)
    at Module._compile (internal/modules/cjs/loader.js:1137:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
    at Module.load (internal/modules/cjs/loader.js:985:32)
    at Function.Module._load (internal/modules/cjs/loader.js:878:14)
    at Module.require (internal/modules/cjs/loader.js:1025:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/zigbee2mqtt-1.17.0/node_modules/zigbee-herdsman-converters/devices.js',
    '/zigbee2mqtt-1.17.0/node_modules/zigbee-herdsman-converters/index.js',
    '/zigbee2mqtt-1.17.0/lib/zigbee.js',
    '/zigbee2mqtt-1.17.0/lib/controller.js',
    '/zigbee2mqtt-1.17.0/index.js'
  ]
}
npm
 ERR! code ELIFECYCLE
npm ERR! errno 1
npm
 ERR! zigbee2mqtt@1.17.0 start: `node index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zigbee2mqtt@1.17.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.`

Clearly something to do with the paths but i don't know how to fix it..or to get the keypad supported by default.

Homeassistant on Debian Linux and Docker ( supervised install)

**Comment by Multipage** [comment]
Finally i got it 'working' But now the only thing that's working:
temperature
linkquality
Link quality (signal strength)

All the rest is N/A under 'Exposes'
zigbee2MQTT says:
ERROR: No converter available for 'voltage' ()
`INFO: MQTT publish: topic 'zigbee2mqtt/0x000d6f0013e97ce5', payload '{"action":"disarm","action_code":"2222","action_zone":23,"linkquality":21,"temperature":17}'
INFO: MQTT publish: topic 'zigbee2mqtt/0x000d6f0013e97ce5', payload '{"action":"","linkquality":21,"temperature":17}'
INFO: MQTT publish: topic 'zigbee2mqtt/0x000d6f0013e97ce5/action', payload 'disarm'
INFO: MQTT publish: topic 'zigbee2mqtt/0x000d6f0013e97ce5', payload '{"linkquality":28,"temperature":17}'`
But data under exposes is N/A or : Unknown acces "r"

**Comment by Multipage** [comment]
I can see a pull request...at zigbee-herdsman-converters
Why isn't it merged yet?

**Reply by Koenkk** [reply]
Can you provide a link to it?

**Reply by Multipage** [reply]
Sure : https://github.com/Koenkk/zigbee-herdsman-converters/pull/1873
But...while i'm running on dev version i can see the code is incompatible with current releases
I tagged you in the PR.

The work must be redone and i can provide  fixes so it 'works' on the dev version but i still need someone to help me with this.
I'm used to script in PHP and this is all new for me :)
I can't do it by myself as i don't know the 'basics' on this and when it's not merged again we will keep going in circles on this one ;)

**Comment by thecobra666** [comment]
I'm able to arm and disarm it with the correct codes but something is off on the **entry and exit_delay action.** I'm unable to stop the beeping with not_ready. The only thing that works is disarm. 

Does any one know the resolution for this? 

{
    "action": "arm_all_zones", // OR "disarm" when being disarmed
    "action_code": "123", // The code being entered
    "action_zone": 0, // The zone being (dis)armed (always 0)
    **"action_transaction": 99 // The transaction number**
}

They mention the action_transaction number, but there is none in the payload coming from the keypad. Yet I do have to push it in the payload to the keypad for it to do arming. 

`debug 2021-06-15 07:49:26: Received Zigbee message from '0x000d6f0014a6fe0a', type 'commandArm', cluster 'ssIasAce', data '{"armmode":3,"code":"XXXX","zoneid":23}' from endpoint 1 with groupID null
info  2021-06-15 07:49:26: MQTT publish: topic 'zigbee2mqtt/0x000d6f0014a6fe0a', payload '{"action":"arm_all_zones","action_code":"XXXX","action_zone":23,"battery":100,"battery_low":false,"contact":true,"linkquality":255,"occupancy":true,"presence":null,"tamper":false,"temperature":24,"voltage":7000}'
info  2021-06-15 07:49:26: MQTT publish: topic 'zigbee2mqtt/0x000d6f0014a6fe0a', payload '{"action":"","action_code":null,"action_zone":null,"battery":100,"battery_low":false,"contact":true,"linkquality":255,"occupancy":true,"presence":null,"tamper":false,"temperature":24,"voltage":7000}'
info  2021-06-15 07:49:26: MQTT publish: topic 'zigbee2mqtt/0x000d6f0014a6fe0a/action', payload 'arm_all_zones'`

When running this keypad in ZHA, they have a dedicated alarm keypad (seen here: https://github.com/home-assistant/core/pull/49080) Would it be possible to create this? 

As last when connected to ZHA and I push on the button to arm a zone and then enter the code. The zone lights up red like it would be armed. How would I do this in zigbee2mqtt?

The arm modes supported are: disarm,arm_day_zones,arm_night_zones,arm_all_zones,exit_delay,entry_delay,not_ready,in_alarm,arming_stay,arming_night,arming_away
